### PR TITLE
Add terraform for AKS

### PR DESCRIPTION
**Pull request by flaviodsr**

### CODE REVIEW

File: infra/terraform/aks/variables.tf [hunk 1]
```diff
@@ -0,0 +1,27 @@
+variable "resource_group_name" {
+  default = "ecosystem-ci"
+}
+
+variable "location" {
+  default = "Germany West Central"
```
Comment (jsuchome) [inline infra/terraform/aks/variables.tf:6]: So ... this is what came naturaly, but didn't this location have problems with Redis Cache creation?

File: infra/terraform/aks/modules/aks/main.tf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  content         = azurerm_kubernetes_cluster.aks.kube_config_raw
+  filename        = "./azurek8s"
+  file_permission = "0600"
+
+  provisioner "local-exec" {
+    command = <<EOT
+      helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+      helm install nginx-ingress ingress-nginx/ingress-nginx --create-namespace \
+        -n ingress-basic --set controller.replicaCount=2
+    EOT
```
Comment (jsuchome) [inline infra/terraform/aks/modules/aks/main.tf:39]: Why are you not using the 'offical' deployment command?
https://kubernetes.github.io/ingress-nginx/deploy/#azure

File: infra/terraform/aks/modules/postgresql/main.tf [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+resource "azurerm_postgresql_database" "databases" {
+  for_each = toset(var.databases)
+
+  name                = each.value
+  resource_group_name = var.resource_group_name
+  server_name         = azurerm_postgresql_server.postgresql.name
+  charset             = "UTF8"
+  collation           = "English_United States.1252"
+}
+
```
Comment (jsuchome) [inline infra/terraform/aks/modules/postgresql/main.tf:38]: Any chance you can also add `schema_migrations` table to `registry` database?

**Comment by jsuchome** [comment]
@flaviodsr so, obviously I'm not terraform expert, so I just wanted to ask, how do you actually execute the creation (or destroy action) and how do you pass the credentials for azure account?

File: infra/terraform/aks/variables.tf (hunk 1 above)
Comment (flaviodsr) [reply infra/terraform/aks/variables.tf:6]: Its not really a problem of the location.
Creating an azure cache deployment does take a considerable amount of time and according to [this feedback post](https://feedback.azure.com/forums/169382-cache/suggestions/7049852-work-on-creation-time-of-redis-cache) the mains reason is:
`"Azure Cache currently runs on Windows VM's and the creation time is largely determined by how fast the underlying VM's become available on-demand."`

File: infra/terraform/aks/modules/aks/main.tf (hunk 2 above)
Comment (flaviodsr) [reply infra/terraform/aks/modules/aks/main.tf:39]: The azure documentation does the ingress deployment using helm, see: https://docs.microsoft.com/en-us/azure/aks/ingress-basic

File: infra/terraform/aks/modules/postgresql/main.tf (hunk 3 above)
Comment (flaviodsr) [reply infra/terraform/aks/modules/postgresql/main.tf:38]: So, besides the database we also need to create this table? According to the Harbor documentation:
`Harbor will create tables automatically when starting up.`

**Comment by flaviodsr** [comment]
@jsuchome The credentials are configured as environment variables (more information at: https://www.terraform.io/docs/providers/azurerm/guides/service_principal_client_secret.html#configuring-the-service-principal-in-terraform).

The creation and destroy are done by running `terraform apply` and `terraform destroy` but before that you need to run `terraform init` to load the resources/modules. Sorry, I should have added a README with those kind of instructions, I just want to get the CI running asap.

File: infra/terraform/aks/modules/postgresql/main.tf (hunk 3 above)
Comment (jsuchome) [reply infra/terraform/aks/modules/postgresql/main.tf:38]: Most of the tables, but not this one I assume. Or why else is it created in the initial set of SQL commands (that are executed by init container for the internal database case)? 

https://github.com/goharbor/harbor/blob/master/make/photon/db/initial-registry.sql#L4

Comment (flaviodsr) [reply infra/terraform/aks/modules/postgresql/main.tf:38]: I believe that unless this is documented we should not add this or we might end up with CI environments different than what customers do.

**Review by dirkmueller — APPROVED** [review]
Looks good to me. Lets merge it and iterate.

**Review by jsuchome — APPROVED** [review]

**Pull request merged**